### PR TITLE
Add comment on multiprocessing.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [x] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Multiprocessing sometimes fails without much clue. 

**Description of the Change:**
Add a warning in the DefaultQubit doc regarding potential erratic behaviour of multiprocessing and a possible to solve it. 

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/devices/default_qubit.py [hunk 1]
```diff
@@ -116,6 +116,24 @@ class DefaultQubit(Device):
...
+        changing the so-called start method. The supported start methods are the following:
+
+        * Windows (win32): spawn (default).
+        * macOS (darwin): spawn (default), fork, forkserver.
+        * Linux (unix): spawn, fork (default), forkserver.
+
+        which can be changed with ``multiprocessing.set_start_method()``. For example,
+        if multiprocessing fails on macOS in your Jupyter notebook environment, try
+        restarting the session, importing multiprocessing and running
+        ``multiprocessing.set_start_method("fork")`` before doing anything.
```
Comment (mudit2812) [inline pennylane/devices/default_qubit.py:132]: It might be better to add this example in a code block.

**Review by mudit2812 — COMMENTED** [review]
Looks good, just one comment. I'll approve after it's addressed.

File: pennylane/devices/default_qubit.py (hunk 1 above)
Comment (vincentmr) [reply pennylane/devices/default_qubit.py]: Like that?
Screenshot from 2023-09-21 15-58-03

File: pennylane/devices/default_qubit.py [hunk 2]
```diff
@@ -116,6 +116,29 @@ class DefaultQubit(Device):
...
+        script with a protected entry point, Jupyter notebook, etc.) This may be solved
+        changing the so-called start method. The supported start methods are the following:
+
+        * Windows (win32): spawn (default).
+        * macOS (darwin): spawn (default), fork, forkserver.
+        * Linux (unix): spawn, fork (default), forkserver.
+
+        which can be changed with ``multiprocessing.set_start_method()``. For example,
+        if multiprocessing fails on macOS in your Jupyter notebook environment, try
+        restarting the session and before doing anything else, type
```
Comment (mudit2812) [inline pennylane/devices/default_qubit.py:131]:
```suggestion
        restarting the session and adding the following to the top:
```

**Review by mudit2812 — APPROVED** [review]

**Comment by timmysilv** [comment]
content looks excellent, but I think `DefaultQubit`'s intro section is getting a bit too lengthy. Can you move the multiprocessing stuff below the backprop stuff, and put it into a drop-down (ps: drop-downs can be made like [this](https://github.com/PennyLaneAI/pennylane/blob/f56e4fe3f0b03ef11b008b247c9b5a5b9ac78a84/pennylane/devices/device_api.py#L61-L62) in sphinx)?

**Review by timmysilv — APPROVED** [review]
🎉 love it :shipit: :shipit:

**Pull request merged**